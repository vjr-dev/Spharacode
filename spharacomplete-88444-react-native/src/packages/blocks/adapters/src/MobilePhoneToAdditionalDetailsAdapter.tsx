import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
