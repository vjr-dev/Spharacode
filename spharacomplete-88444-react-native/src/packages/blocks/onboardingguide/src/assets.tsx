export const onBoardingOne = require("../assets/Onboarding_One.png");
export const onBoardingTwo = require("../assets/Onboarding_Two.png");
export const onboardingThree = require("../assets/Onboarding_Three.png");
export const onboardingFour = require("../assets/Onboarding_Four.png");
export const imgPasswordVisible = require("../assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../assets/ic_password_invisible.png");
