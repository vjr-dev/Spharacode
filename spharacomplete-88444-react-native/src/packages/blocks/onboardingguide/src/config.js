Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.title1="Sell or Buy products";
exports.title2="Sell / Buy products";
exports.title3=" Buy products";
exports.title4="Sell products";

exports.subtitle1="Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper";
exports.subtitle2="Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native";
exports.subtitle3="Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper";
exports.subtitle4="Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper";
// Customizable Area End