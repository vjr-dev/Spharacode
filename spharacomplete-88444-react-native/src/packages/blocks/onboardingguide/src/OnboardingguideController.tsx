import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { runEngine } from "../../../framework/src/RunEngine";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
// Customizable Area Start
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class OnboardingguideController extends BlockComponent<
  Props,
  S,
  SS
> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [];

    this.state = {};
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);
  }

  // Customizabele Area Start
  skipOnboarding() {
    const msg: Message = new Message(getName(MessageEnum.OnboardingGuideControllerSkipOnboarding));
    msg.addData(getName(MessageEnum.NavigationPropsMessage), this.props);
    this.send(msg);
  }

  exitOnboarding() {
    const msg: Message = new Message(getName(MessageEnum.OnboardingGuideControllerDoneOnboarding));
    msg.addData(getName(MessageEnum.NavigationPropsMessage), this.props);
    this.send(msg);
  }
  // Customizable Area End
}
