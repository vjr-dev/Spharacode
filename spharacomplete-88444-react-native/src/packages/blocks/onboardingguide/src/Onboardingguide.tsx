import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  Image,
  TouchableOpacity,
  View,
  SafeAreaView
} from "react-native";
import {
  onBoardingOne,
  onboardingThree,
  onBoardingTwo,
  onboardingFour
} from "../src/assets";
import Onboarding from "react-native-onboarding-swiper";
// Customizable Area End
import OnboardingguideController, {
  Props,
  configJSON
} from "./OnboardingguideController";

export default class Onboardingguide extends OnboardingguideController {
  constructor(props: Props) {
    super(props);
  }

  // Customizable Area Start
  Done = () => (
    <TouchableOpacity
      testID="btnDone"
      onPress={() => this.exitOnboarding()}
      style={{ alignSelf: "flex-end", marginRight: 20 }}
    >
      <Text style={{ fontSize: 16, fontWeight: "100" }}>Done</Text>
    </TouchableOpacity>
  );
  // Customizable Area Done

  render() {
    return (
      // Customizable Area Start
      <Onboarding
        onSkip={() => this.skipOnboarding()}
        onDone={() => this.exitOnboarding()}
        DoneButtonComponent={this.Done}
        containerStyles={{
          justifyContent: "flex-start",
          alignItems: "flex-start"
        }}
        pages={[
          {
            backgroundColor: "white",
            image: (
              <SafeAreaView style={styles.container}>
                <View style={styles.parent}>
                  <Image
                    source={onBoardingOne}
                    resizeMode={"contain"}
                    style={styles.imageStyle}
                  />
                </View>
                <View style={styles.ovalShape} />
              </SafeAreaView>
            ),
            title: (
              <View style={styles.title}>
                <Text style={styles.textStyle}>{configJSON.title1}</Text>
              </View>
            ),

            subtitle: configJSON.subtitle1,
            titleStyles: { color: "black", fontSize: 20 }
          },
          {
            backgroundColor: "white",
            image: (
              <SafeAreaView style={styles.container}>
                <View style={styles.parent}>
                  <Image
                    source={onBoardingTwo}
                    resizeMode={"contain"}
                    style={styles.imageStyle}
                  />
                </View>
                <View style={styles.ovalShape} />
              </SafeAreaView>
            ),
            title: (
              <View style={styles.title}>
                <Text style={styles.textStyle}>{configJSON.title2}</Text>
              </View>
            ),
            subtitle: configJSON.subtitle2,
            titleStyles: { color: "black", fontSize: 20 }
          },

          {
            backgroundColor: "white",
            image: (
              <SafeAreaView style={styles.container}>
                <View style={styles.parent}>
                  <Image
                    source={onboardingThree}
                    resizeMode={"contain"}
                    style={styles.imageStyle}
                  />
                </View>
                <View style={styles.ovalShape} />
              </SafeAreaView>
            ),
            title: (
              <View style={styles.title}>
                <Text style={styles.textStyle}>{configJSON.title3}</Text>
              </View>
            ),
            subtitle: configJSON.subtitle3,
            titleStyles: { color: "black", fontSize: 20 }
          },

          {
            backgroundColor: "white",
            image: (
              <SafeAreaView style={styles.container}>
                <View style={styles.parent}>
                  <Image
                    source={onboardingFour}
                    resizeMode={"contain"}
                    style={styles.imageStyle}
                  />
                </View>
                <View style={styles.ovalShape} />
              </SafeAreaView>
            ),
            title: (
              <View style={styles.title}>
                <Text style={styles.textStyle}>{configJSON.title4}</Text>
              </View>
            ),
            subtitle: configJSON.subtitle4,
            titleStyles: { color: "black", fontSize: 20 }
          }
        ]}
      />
      // Customizable Area End
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    width: "100%",
    backgroundColor: "blue"
  },

  title: {
    alignSelf: "center",
    padding: 15
  },

  parent: {
    backgroundColor: "cornflowerblue",
    width: "100%",
    alignSelf: "center"
  },

  imageStyle: {
    height: 280,
    width: "90%",
    alignSelf: "center",
    padding: 20,
    marginTop: 40
  },

  ovalShape: {
    width: 200,
    height: 200,
    backgroundColor: "cornflowerblue",
    borderRadius: 100,
    alignSelf: "center",
    transform: [{ scaleX: 3 }],
    position: "absolute",
    bottom: -50,
    zIndex: -1
  },

  textStyle: {
    fontSize: 18,
    fontWeight: "bold"
  }
});
// Customizable Area End
