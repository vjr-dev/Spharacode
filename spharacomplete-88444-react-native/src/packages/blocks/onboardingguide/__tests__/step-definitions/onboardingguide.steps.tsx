import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'

import * as helpers from '../../../../framework/src/Helpers'
// import {runEngine} from '../../../../framework/src/RunEngine'
// import {Message} from "../../../../framework/src/Message"

// import MessageEnum, {getName} from "../../../../framework/src/Messages/MessageEnum"; 
import React from "react";
import Onboardingguide from "../../src/Onboardingguide"
const navigation = require("react-navigation")  

const screenProps = {
    navigation: navigation,
    id: "Onboardingguide"
  }    

const feature = loadFeature('./__tests__/features/onboardingguide-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    }); 

    test('User navigates to onboardingguide', ({ given, when, then }) => {
      let OnboardingGuideWrapper:ShallowWrapper;
        let instance:Onboardingguide; 

        given('I am a User loading onboardingguide', () => {
          OnboardingGuideWrapper= shallow(<Onboardingguide {...screenProps}/>)
          expect(OnboardingGuideWrapper).toBeTruthy();
          expect(OnboardingGuideWrapper).toMatchSnapshot();
        });

        when('I navigate to the onboardingguide', () => {
           
             instance = OnboardingGuideWrapper.instance()as Onboardingguide;
           
        });

        then('onboardingguide will load with out errors', () => {
            expect(OnboardingGuideWrapper).toBeTruthy()
            expect(OnboardingGuideWrapper).toMatchSnapshot()

            
        });

        then('I can leave the screen with out errors', () => {
            instance.componentWillUnmount()
            expect(OnboardingGuideWrapper).toBeTruthy()
            expect(OnboardingGuideWrapper).toMatchSnapshot()
        });
    });


});
