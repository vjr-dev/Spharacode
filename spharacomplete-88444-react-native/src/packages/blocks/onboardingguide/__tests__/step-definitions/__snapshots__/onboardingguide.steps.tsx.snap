// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onboardingguide User navigates to onboardingguide 1`] = `
<Onboarding
  DoneButtonComponent={[Function]}
  DotComponent={[Function]}
  NextButtonComponent={[Function]}
  SkipButtonComponent={[Function]}
  allowFontScalingButtons={true}
  allowFontScalingText={true}
  bottomBarColor="transparent"
  bottomBarHeight={60}
  bottomBarHighlight={true}
  containerStyles={
    Object {
      "alignItems": "flex-start",
      "justifyContent": "flex-start",
    }
  }
  controlStatusBar={true}
  imageContainerStyles={null}
  nextLabel="Next"
  onDone={[Function]}
  onSkip={[Function]}
  pageIndexCallback={null}
  pages={
    Array [
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_One.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell or Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Two.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell / Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Three.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
             Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Four.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
    ]
  }
  showDone={true}
  showNext={true}
  showPagination={true}
  showSkip={true}
  skipLabel="Skip"
  skipToPage={null}
  subTitleStyles={null}
  titleStyles={null}
  transitionAnimationDuration={500}
/>
`;

exports[`onboardingguide User navigates to onboardingguide 2`] = `
<Onboarding
  DoneButtonComponent={[Function]}
  DotComponent={[Function]}
  NextButtonComponent={[Function]}
  SkipButtonComponent={[Function]}
  allowFontScalingButtons={true}
  allowFontScalingText={true}
  bottomBarColor="transparent"
  bottomBarHeight={60}
  bottomBarHighlight={true}
  containerStyles={
    Object {
      "alignItems": "flex-start",
      "justifyContent": "flex-start",
    }
  }
  controlStatusBar={true}
  imageContainerStyles={null}
  nextLabel="Next"
  onDone={[Function]}
  onSkip={[Function]}
  pageIndexCallback={null}
  pages={
    Array [
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_One.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell or Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Two.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell / Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Three.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
             Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Four.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
    ]
  }
  showDone={true}
  showNext={true}
  showPagination={true}
  showSkip={true}
  skipLabel="Skip"
  skipToPage={null}
  subTitleStyles={null}
  titleStyles={null}
  transitionAnimationDuration={500}
/>
`;

exports[`onboardingguide User navigates to onboardingguide 3`] = `
<Onboarding
  DoneButtonComponent={[Function]}
  DotComponent={[Function]}
  NextButtonComponent={[Function]}
  SkipButtonComponent={[Function]}
  allowFontScalingButtons={true}
  allowFontScalingText={true}
  bottomBarColor="transparent"
  bottomBarHeight={60}
  bottomBarHighlight={true}
  containerStyles={
    Object {
      "alignItems": "flex-start",
      "justifyContent": "flex-start",
    }
  }
  controlStatusBar={true}
  imageContainerStyles={null}
  nextLabel="Next"
  onDone={[Function]}
  onSkip={[Function]}
  pageIndexCallback={null}
  pages={
    Array [
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_One.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell or Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Two.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell / Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Three.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
             Buy products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
      Object {
        "backgroundColor": "white",
        "image": <SafeAreaView
          style={
            Object {
              "backgroundColor": "blue",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/onboardingguide/assets/Onboarding_Four.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 280,
                  "marginTop": 40,
                  "padding": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "cornflowerblue",
                "borderRadius": 100,
                "bottom": -50,
                "height": 200,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "scaleX": 3,
                  },
                ],
                "width": 200,
                "zIndex": -1,
              }
            }
          />
        </SafeAreaView>,
        "subtitle": "Done with React Native Onboarding Swiper Done with React Native Onboarding SwiperDone with React Native Onboarding SwiperDone with React Native Onboarding Swiper",
        "title": <View
          style={
            Object {
              "alignSelf": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Sell products
          </Text>
        </View>,
        "titleStyles": Object {
          "color": "black",
          "fontSize": 20,
        },
      },
    ]
  }
  showDone={true}
  showNext={true}
  showPagination={true}
  showSkip={true}
  skipLabel="Skip"
  skipToPage={null}
  subTitleStyles={null}
  titleStyles={null}
  transitionAnimationDuration={500}
/>
`;
